evm_loader/program/src/instruction/neon_tokens_deposit.rs

```diff
@@ -167,16 +167,13 @@ fn execute(
     let deposit = U256::from(accounts.source.delegated_amount) * 10_u128.pow(additional_decimals);
     let mut ethereum_account =
         EthereumAccount::from_account(program_id, accounts.ethereum_account)?;
-    ethereum_account.balance = ethereum_account
-        .balance
-        .checked_add(deposit)
-        .ok_or_else(|| {
-            E!(
-                ProgramError::InvalidArgument;
-                "Account {} - balance overflow",
-                ethereum_address
-            )
-        })?;
+    ethereum_account.balance = ethereum_account.balance.checked_add(deposit).ok_or({
```

Comment:
You have introduced a bug by this change. This will log error every time, even if no error happen. Look into `E!` macro.

Reply:
Interesting, I guess it was one of the clippy lints. I'll check it. Well spotted!

Reply:
You still has other similar code with the same issue. Revert it.
Also do checks with clippy v1.64